src/olympia/devhub/views.py

```diff
@@ -402,6 +402,10 @@ def disable(request, addon_id, addon):
 @post_required
 def unlist(request, addon_id, addon):
     addon.update(is_listed=False, disabled_by_user=False)
+    # In https://github.com/mozilla/addons-server/issues/3471 this view will
+    # no longer be global, but in the meantime we need to set the channel
+    # property on all versions to stay consistent.
+    addon.versions.update(channel=amo.RELEASE_CHANNEL_UNLISTED)
```

Comment:
Aaaand it's a good thing you asked. I double-checked and for *many to many* relationships, we hide the deleted stuff, on purpose. I'll add a test and fix that.